frontend/src/app/patients/PatientForm.tsx

```diff
@@ -121,38 +122,31 @@ const UPDATE_PATIENT = gql`
   }
 `;
 
-const Fieldset = (props) => (
-  <fieldset className="prime-fieldset">
-    <legend>
-      <h3>{props.legend}</h3>
-    </legend>
-    {props.children}
-  </fieldset>
-);
+interface Props {
+  activeFacilityId: string;
+  patientId?: string;
+  patient?: any; //TODO: TYPES
+}
 
-// TODO: move to an Address component
-const stateChoices = [
-  { label: "- Select - ", value: "" },
-  ...stateCodes.map((c) => ({ label: c, value: c })),
-];
-
-const PatientForm = (props) => {
+const PatientForm = (props: Props) => {
   const appInsights = useAppInsightsContext();
-  const trackAddPatient = useTrackEvent(appInsights, "Add Patient");
-  const trackUpdatePatient = useTrackEvent(appInsights, "Update Patient");
+  const trackAddPatient = useTrackEvent(appInsights, "Add Patient", {});
+  const trackUpdatePatient = useTrackEvent(appInsights, "Update Patient", {});
 
   const [addPatient] = useMutation(ADD_PATIENT);
   const [updatePatient] = useMutation(UPDATE_PATIENT);
   const [formChanged, setFormChanged] = useState(false);
   const [patient, setPatient] = useState(props.patient);
   const [submitted, setSubmitted] = useState(false);
 
-  const onChange = (e) => {
+  const onChange = (
+    e: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>
+  ) => {
     let value = e.target.value;
     if (e.target.type === "checkbox") {
       value = {
         ...patient[e.target.name],
-        [e.target.value]: e.target.checked,
+        [e.target.value]: (e.target as any).checked,
```

Comment:
Typescript didn't understand that the guard a couple of lines above guaranteed that the `checked` property would exist, it was afraid this would be a select.

Reply:
```
    let value = e.target.value;
    let target = e.target;
    if (target.type === "checkbox") {
      value = {
        ...patient[e.target.name],
        [e.target.value]: target .checked,
```
^ this should make TypeScipt happy